eng/Subsets.props

```diff
@@ -276,7 +276,7 @@
     <CrossDacHostArch Condition="'$(TargetArchitecture)' == 'arm'">x86</CrossDacHostArch>
   </PropertyGroup>
 
-  <ItemGroup Condition="$(_subset.Contains('+linuxdac+')) and $([MSBuild]::IsOsPlatform(Windows)) and '$(TargetArchitecture)' != 'x86'">
+  <ItemGroup Condition="$(_subset.Contains('+linuxdac+')) and $([MSBuild]::IsOsPlatform(Windows)) and '$(BuildArchitecture)' != '$(TargetArchitecture)' and ('$(BuildArchitecture)' != 'x64' or '$(TargetArchitecture)' != 'x86')">
```

Comment:
This isn't correct - x64 Windows will build the x64 linux crossdac.

Reply:
Have had another go at this -- this time kept the check identical but excluded architectures other than x64 and x86. Since L274-277 seem to assume this is the case anyway, I've made the assumption this will be sufficient, but I'm unsure in exactly which scenarios this should be running.

Let me know if this isn't correct, and I'll try and correct again. Apologies.

Reply:
No need to apologize. I should have also been more descriptive. This won't break the official build. That being said - it's something that we should fix at some point - so that it's possible to use arm64 windows debuggers to analyze linux dumps.